announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java

```diff
@@ -1426,16 +1426,7 @@ private boolean isOkayToDisplayMessageMenu(AnnouncementActionState state)
 	 */
 	private boolean isOkToShowOptionsButton(String statusName)
 	{
-		if (SiteService.allowUpdateSite(ToolManager.getCurrentPlacement().getContext()) && !isOnWorkspaceTab())
-		{
-			return (statusName == null || statusName.equals(CANCEL_STATUS) || statusName.equals(POST_STATUS)
-					|| statusName.equals(DELETE_ANNOUNCEMENT_STATUS) || statusName.equals(FINISH_DELETING_STATUS)
-					|| statusName.equals("noSelectedForDeletion") || statusName.equals(NOT_SELECTED_FOR_REVISE_STATUS));
-		}
-		else
-		{
-			return false;
-		}
+		return SiteService.allowUpdateSite(ToolManager.getCurrentPlacement().getContext()) && !isOnWorkspaceTab();
```

Comment:
Yep. You always want the tabs to be showing, as long as you have the appropriate permission. You shouldn't have tabs popping into and out of existence depending on what view state you are in - they should just be fixed with just current tab being indicated. All those STATUS things are view states.